### ISSUE TITLE
Added support for detecting issues with BytesMarshallable

### ISSUE THREAD

**Issue by peter-lawrey** [OpenHFT/Chronicle-Bytes#98]
BytesMarshallable can be hard to debug with no meta data.
This adds an option to enable meta data compatible with BinaryWire so the stream can be checked
```java
assert NativeBytes.setNewGuarded(true);
```
or
```
-Dbytes.guarded=true
```
This changes the format and makes the data larger but adds checks which makes it easier to find inconsistencies between WriteMarshalable and ReadMarshallable.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.17.23](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.17.23), [BOM-2.17.233](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.233)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.17.23](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.17.23), [BOM-2.17.241](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.241)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.20.101](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
